cas-server-webapp/src/main/java/org/jasig/cas/support/pac4j/authentication/handler/support/ClientAuthenticationHandler.java

```diff
@@ -56,6 +56,17 @@ public ClientAuthenticationHandler(final Clients theClients) {
 
     /**
      * {@inheritDoc}
+     *
+     * Note on customization:
```

Comment:
Here is how and why we customize the principal building process provided by pac4j.